### PR TITLE
ci: release v2 workflow

### PR DESCRIPTION
**Pull request by talkor**
https://monday.monday.com/boards/3532714909/pulses/7202842195

### CODE REVIEW

File: .github/workflows/release-v2.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        with:
+          npm_token: ${{ secrets.npm_token }}
+      - uses: ./.github/actions/git-creds
+      - name: Check current branch is "version2"
+        run: |
+          if [[ "$(git branch --show-current)" != "version2" ]]; then
+            echo "Error: Not on 'version2' branch."
+            exit 1
+          fi
+      - name: Build
```
Comment (talkor) [inline .github/workflows/release-v2.yml:28]: Lerna doesn't support version/publish for a single package so we just need to do all the checks as usual without lerna

File: .github/workflows/release-v2.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        run: yarn stylelint
+      - name: Generate new core version
+        run: |
+          npm version patch -m "chore(release): bump core package version to %s"
+      - run: yarn config set registry https://registry.npmjs.org/
+      - name: Setup .npmrc for publish
+        id: setup-npmrc
+        run: echo "//registry.npmjs.org/:_authToken=$NODE_AUTH_TOKEN" > .npmrc
+      - name: Publish core to npm
+        run: npm publish --dry-run
```
Comment (talkor) [inline .github/workflows/release-v2.yml:44]: Will be removed after dry-running on the actual branch first time to make sure it's all ok

File: .github/workflows/release-v2.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Run Setup
+        uses: ./.github/actions/setup
+        with:
+          npm_token: ${{ secrets.npm_token }}
+      - uses: ./.github/actions/git-creds
+      - name: Check current branch is "version2"
```
Comment (talkor) [inline .github/workflows/release-v2.yml:22]: Just for safety

Comment (YossiSaadi) [reply .github/workflows/release-v2.yml:22]: see top comment regarding `if` for `job` level

File: .github/workflows/release-v2.yml [hunk 4]
```diff
@@ -0,0 +1,47 @@
+name: Release v2 version
+
+on: workflow_dispatch
+
+jobs:
+  release:
```
[the remark this answers is not in the document]
Comment (YossiSaadi) [reply .github/workflows/release-v2.yml:6]: this can also work:
```github.ref_name == 'version2'```

File: .github/workflows/release-v2.yml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+            exit 1
+          fi
+      - name: Build
+        run: yarn build
+      - name: Lint
+        run: yarn lint
+      - name: Test
+        run: yarn test
+      - name: Stylelint
+        run: yarn stylelint
```
Comment (YossiSaadi) [inline .github/workflows/release-v2.yml:35]: why don't you use the `build-and-upload` and `test` jobs before running `release` job instead? as it is in the release.yml now

Comment (YossiSaadi) [reply .github/workflows/release-v2.yml:35]: oh they're dependent on lerna...

File: .github/workflows/release-v2.yml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+      - name: Generate new core version
+        run: |
+          npm version patch -m "chore(release): bump core package version to %s"
```
Comment (YossiSaadi) [inline .github/workflows/release-v2.yml:38]: where does the `%s` comes from?

**Review by YossiSaadi — APPROVED** [review]
looks cool, had a few minor suggestion

File: .github/workflows/release-v2.yml (hunk 6 above)
Comment (talkor) [reply .github/workflows/release-v2.yml:38]: https://docs.npmjs.com/cli/v8/commands/npm-version#description

File: .github/workflows/release-v2.yml (hunk 3 above)
Comment (talkor) [reply .github/workflows/release-v2.yml:22]: Removed this

File: .github/workflows/release-v2.yml (hunk 6 above)
Comment (YossiSaadi) [reply .github/workflows/release-v2.yml:33]: cool!

**Pull request merged**